nixos-test.nix

```
@@ -0,0 +1,26 @@
+{ hello, testers, nix-serve }:
+
+testers.runNixOSTest {
```

You could move the invocation of the test to `flake.nix` so that this file can be copied back and forth between here and NixOS.
I guess you'll need to either use an overlay then, or merge https://github.com/NixOS/nixpkgs/pull/299542 so that you can call `extendNixOS` to add the `package = nix-serve;` part.

```suggestion
{
```

---

Wait no, that's not even necessary. Just call it as

```nix
# flake.nix
testers.runNixOSTest {
  imports = [ ./nixos-test.nix ];
  extraBaseModules = {
    services.nix-serve.package = self.defaultPackage.${system};
  };
}
```

---

Applied.